packages/core/review-workflows/admin/src/routes/settings/components/Stages.tsx

```diff
@@ -517,25 +517,34 @@ const PermissionsField = ({ disabled, name, placeholder, required }: Permissions
   const { value = [], error, onChange } = useField<StagePermission[]>(name);
   const allStages = useForm<WorkflowStage[]>('PermissionsField', (state) => state.values.stages);
   const onFormValueChange = useForm('PermissionsField', (state) => state.onChange);
+  const rolesErrorCount = React.useRef(0);
 
-  const { data: roles = [], isLoading } = useGetAdminRolesQuery();
+  const { data: roles = [], isLoading, error: getRolesError } = useGetAdminRolesQuery();
 
   // Super admins always have permissions to do everything and therefore
   // there is no point for this role to show up in the role combobox
   const filteredRoles = roles?.filter((role) => role.code !== 'strapi-super-admin') ?? [];
 
   React.useEffect(() => {
-    if (!isLoading && roles.length === 0) {
+    if (
+      !isLoading &&
+      getRolesError &&
+      'status' in getRolesError &&
+      getRolesError.status == 403 &&
+      rolesErrorCount.current === 0
+    ) {
+      rolesErrorCount.current = 1;
+
```

Comment:
@alexandrebodin I'm not able to reproduce that last fix

https://github.com/strapi/strapi/assets/8087692/0470c565-32e1-4f2c-b55d-37f8842602d0

Reply:
if you remove the permissions to read roles in your test role you should get it